### PR TITLE
Add parallel_scan exercise

### PR DESCRIPTION
**Pull request by masterleinad**
The tutorial slides are in https://github.com/kokkos/Tutorials/pull/48.

### CODE REVIEW

File: Exercises/parallel_scan/Solution/parallel_scan.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  using value_type = ValueType;
+
+  Factorial(Kokkos::View<value_type *> view) : m_view(view) {}
+
+  KOKKOS_FUNCTION void init(value_type &value) const { value = 1.; }
+  KOKKOS_FUNCTION void join(value_type &lhs, const value_type &rhs) const {
+    lhs *= rhs;
+  }
+
+  KOKKOS_FUNCTION void operator()(int i, value_type &update,
```
Comment (dalg24) [inline Exercises/parallel_scan/Solution/parallel_scan.cpp:30]: maybe update -> partial_product

File: Exercises/parallel_scan/Solution/parallel_scan.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <cstdio>
```
Comment (dalg24) [inline Exercises/parallel_scan/Solution/parallel_scan.cpp:18]: You need to include `<iostream>`
I expect this does not compile with 4.3

File: Exercises/parallel_scan/Begin/parallel_scan.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <Kokkos_Core.hpp>
+#include <cstdio>
+
+template <typename ValueType> struct Factorial {
+  /* FIXME */
```
Comment (dalg24) [inline Exercises/parallel_scan/Begin/parallel_scan.cpp:21]: Didn't we commonly use EXERCISE to mark places that needed to be edited?

File: Exercises/parallel_scan/Begin/parallel_scan.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  Kokkos::initialize(argc, argv);
+  {
+    int n = 10;
+    Kokkos::View<double *> view("view", n);
+
+    Kokkos::parallel_scan(n, Factorial{view});
+
+    auto host_view =
+        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, view);
+    Kokkos::View<double *> reference_view("reference", n);
```
Comment (crtrott) [inline Exercises/parallel_scan/Begin/parallel_scan.cpp:34]:
```suggestion
    Kokkos::View<double *, Kokkos::HostSpace> reference_view("reference", n);
```

File: Exercises/parallel_scan/Solution/parallel_scan.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  Kokkos::initialize(argc, argv);
+  {
+    int n = 10;
+    Kokkos::View<double *> view("view", n);
+
+    Kokkos::parallel_scan(n, Factorial{view});
+
+    auto host_view =
+        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, view);
+    Kokkos::View<double *> reference_view("reference", n);
```
Comment (crtrott) [inline Exercises/parallel_scan/Solution/parallel_scan.cpp:51]:
```suggestion
    Kokkos::View<double *, Kokkos::HostSpace> reference_view("reference", n);
```

**Review by crtrott — CHANGES_REQUESTED** [review]
Two mistakes: did you try this with GPU?

**Pull request merged**